tinygrad/renderer/assembly.py

```diff
@@ -216,6 +216,14 @@ def cast(a, dtype:DType, atype:DType, bitcast=False, u=None, pred=False):
         if lang.load_global:
           dt = dtypes.ulong if dtype.__class__ == PtrDType else dtype
           kk(*lang.render_load(args[1], ssa(u, 'dat', dtype=lang.types[dt]), dt, ss=".param"))
+      elif uop is UOps.WMMA:
+        wmma=[]
+        for vv in vin[:2]:
+          for i in range(0, len(r[vv]), 2):
+            wmma.append(ssa(None, "wmma", "b32"))
+            kk(f'mov.b32 {wmma[-1]}, {{{", ".join(r[vv][i:i+2])}}};')
```

Comment:
Hmm, does there have to be an extra mov here, or is there a way plan this earlier and have the data already be in the right place?

Reply:
Not reliably afaik, what this code actually boils down into is this. 
```
     ld.global.b16   %val_f16_0, [%alu_u64_0+0];
     ...
     ld.global.b16   %val_f16_1, [%alu_u64_1+0];
     ...
     mov.b32         %wmma_b32_0, {%val_f16_0, %val_f16_1};
```
theoretically it is possible to load directly like this 
`     ld.global.b16   %wmma_b32_0, [%alu_u64_0+0];`
But in practice `%val_f16_0` and `%val_f16_1` aren't always contiguous and we fail with bad memory access